### PR TITLE
Addition of test data in test_helper file.

### LINKED ISSUE #30: 19. Admin Dashboard
As an admin,
When I visit the admin dashboard (/admin)
Then I see a header indicating that I am on the admin dashboard

### PR DESCRIPTION
**Pull request by bserge13**
Taking advantage of test_helper file with addition of both test_data and data for "best __" stories.

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -5,4 +5,6 @@ class Invoice < ApplicationRecord
   has_many :items, through: :invoice_items
 
   enum status: ["completed", "cancelled", "in progress"]
+
+  validates :status, presence: true
```
Comment (Sykogst) [inline app/models/invoice.rb]: I think enums have a more specific shouldamatcher than can be used for validations, allows checking of values too.

it { should define_enum_for(:status).with_values([completed: 0, cancelled: 1, in_progress: 2]) }

not 100% sure how the space is handled with the "in progress"

File: spec/models/invoice_item_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+RSpec.describe InvoiceItem do 
+  describe "relationships" do 
+    it { should belong_to :invoice }
+    it { should belong_to :item }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :quantity }
+    it { should validate_presence_of :unit_price }
+    it { should validate_presence_of :status }
```
Comment (Sykogst) [inline spec/models/invoice_item_spec.rb:12]: Same for here with the enums, something like

it { should define_enum_for(:status).with_values([pending: 0, packaged: 1, shipped: 2]) }

File: spec/models/invoice_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+RSpec.describe Invoice do 
+  describe "relationships" do 
+    it { should belong_to :customer }
+    it { should have_many :transactions }
+    it { should have_many :invoice_items }
+    it { should have_many(:items).through(:invoice_items) }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :status }
```
Comment (Sykogst) [inline spec/models/invoice_spec.rb:12]: Same for here with the enums, something like

it { should define_enum_for(:status).with_values([completed: 0, cancelled: 1, in_progress: 2]) }

unsure on how the space is handled for "in progress"

File: spec/models/item_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    it { should belong_to :merchant }
+    it { should have_many :invoice_items }
+    it { should have_many(:invoices).through(:invoice_items) }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :name }
+    it { should validate_presence_of :description }
+    it { should validate_presence_of :unit_price }
+    it { should validate_presence_of :status }
```
Comment (Sykogst) [inline spec/models/item_spec.rb:14]: Same for here with the enums, something like

it { should define_enum_for(:status).with_values([disabled: 0, enabled: 1]) }

File: spec/models/merchant_spec.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+RSpec.describe Merchant do 
+  describe "relationships" do 
+    it { should have_many :items }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :name }
+    it { should validate_presence_of :status }
```
Comment (Sykogst) [inline spec/models/merchant_spec.rb:10]: Same for here with the enums, something like

it { should define_enum_for(:status).with_values([disabled: 0, enabled: 1]) }

File: spec/models/transaction_spec.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+RSpec.describe Transaction do 
+  describe "relationships" do 
+    it { should belong_to :invoice }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :credit_card_number }
+    it { should validate_presence_of :credit_card_expiration_date }
+    it { should validate_presence_of :result }
```
Comment (Sykogst) [inline spec/models/transaction_spec.rb:11]: Same for here with the enums, something like

it { should define_enum_for(:status).with_values([success: 0, failed: 1]) }

File: app/models/invoice.rb (hunk 1 above)
Comment (bserge13) [reply app/models/invoice.rb]: Good callout! I'll look into it and add a little more robustness to the test. Thanks Sam!

File: spec/models/transaction_spec.rb (hunk 6 above)
Comment (trpubz) [reply spec/models/transaction_spec.rb:11]: completed, no `[]`

File: spec/models/merchant_spec.rb (hunk 5 above)
Comment (trpubz) [reply spec/models/merchant_spec.rb:10]: completed, no `[]`

File: spec/models/item_spec.rb (hunk 4 above)
Comment (trpubz) [reply spec/models/item_spec.rb:14]: completed, no `[]`

File: spec/models/invoice_spec.rb (hunk 3 above)
Comment (trpubz) [reply spec/models/invoice_spec.rb:12]: completed, no `[]`

File: spec/models/invoice_item_spec.rb (hunk 2 above)
Comment (trpubz) [reply spec/models/invoice_item_spec.rb:12]: completed, no `[]`

File: app/models/invoice.rb (hunk 1 above)
Comment (trpubz) [reply app/models/invoice.rb]: see latest commit to add sam's suggestions, no `[]`

**Pull request merged**